src/lib/core/datetime/native-date-adapter.ts

```diff
@@ -192,6 +197,16 @@ export class NativeDateAdapter extends DateAdapter<Date> {
     ].join('-');
   }
 
+  isValidDate(value: any) {
+    if (value == null) {
```

Comment:
@jelbourn It kind of feels wrong that people extending `DateAdapter` even have the ability to say "null is invalid" maybe an indicator that I should move all the null checks into the datepicker code itself, WDYT?